backend/src/main/java/trn/logistics/knapsack/dto/SolutionRequest.java

```diff
@@ -0,0 +1,21 @@
+package trn.logistics.knapsack.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collection;
+
+@Getter
+@Slf4j
+@AllArgsConstructor
+public class SolutionRequest {
+
+    private Long id;
+
+    private Collection<Long> materialIdCollection;
```

Comment:
```suggestion
    @NotNull
    private Collection<Long> materialIds;
```

See:
https://stackoverflow.com/questions/45867473/should-you-use-notnull-and-jsonpropertyrequired-in-the-same-object
https://stackoverflow.com/questions/12813319/enforce-not-null-field-in-json-object/23133086
http://robaustin.wikidot.com/annotations-and-notnull